lib/rtp_h264/depayloader.ex

```diff
@@ -30,38 +30,38 @@ defmodule Membrane.RTP.H264.Depayloader do
   end
 
   @impl true
-  def handle_init(_opts) do
-    {:ok, %State{}}
+  def handle_init(_ctx, _opts) do
+    {[], %State{}}
   end
 
   @impl true
-  def handle_caps(:input, _caps, _context, state) do
-    caps = %RemoteStream{content_format: H264, type: :packetized}
-    {{:ok, caps: {:output, caps}}, state}
+  def handle_stream_format(:input, _stream_format, _context, state) do
+    stream_format = %RemoteStream{content_format: H264, type: :packetized}
+    {[stream_format: {:output, stream_format}], state}
   end
 
   @impl true
   def handle_process(:input, %Buffer{payload: ""}, _ctx, state) do
     Membrane.Logger.debug("Received empty RTP packet. Ignoring")
-    {:ok, state}
+    {[], state}
   end
 
   @impl true
   def handle_process(:input, buffer, _ctx, state) do
     with {:ok, {header, _payload} = nal} <- NAL.Header.parse_unit_header(buffer.payload),
          unit_type = NAL.Header.decode_type(header),
-         {{:ok, actions}, state} <- handle_unit_type(unit_type, nal, buffer, state) do
-      {{:ok, actions}, state}
+         {actions, state} when is_list(actions) <- handle_unit_type(unit_type, nal, buffer, state) do
```

Comment:
I *really* don't like that guard. In fact, I think I would keep the return of `handle_unit_type` as it was. It's not a Membrane callback and it definitely needs to be able to return an error. A standard way to do it is `{:ok, something} | {:error, reason}` and I would stick to that.